worldguard-bukkit/src/main/java/com/sk89q/worldguard/bukkit/WorldGuardPlugin.java

```diff
@@ -205,10 +210,12 @@ public void onEnable() {
         }
         worldListener.registerEvents();
 
-        Bukkit.getScheduler().runTask(this, () -> {
+        getScheduler().runAsync(() -> {
```

Comment:
This runAsync call is not needed. You already create a task for every player in the for loop.

Reply:
this actually presents a race condition as running the init on region threads could actually happen after the async sessionmanager loop, leading to a player getting double-initialized.
that is, the bukkitsessionmanager itself should be handling scheduling for the individual player "events" (process/run)

Reply:
@wizjany, creating a session does the initialization, so I don't see the need to call the initialization method in https://github.com/EngineHub/WorldGuard/blob/78498491ddf94f2fce376614e17556e5ff8f6327/worldguard-bukkit/src/main/java/com/sk89q/worldguard/bukkit/session/BukkitSessionManager.java#L61 so can try to get a session to pre-create it.